plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -896,7 +899,50 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
         return Optional.empty();
     }
 
-    private static void appendInitialTableEntries(
+    @Override
+    public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnMetadata newColumnMetadata)
+    {
+        if (newColumnMetadata.getComment() != null) {
```

Comment:
Spark's Delta can cope with columns comments.

```
spark-sql> alter table t1 add columns (z string comment 'z comment');
Time taken: 1.54 seconds
spark-sql> desc t1;
x	int	
y	string	
z	string	z comment
		
# Partitioning		
Not partitioned		
Time taken: 0.075 seconds, Fetched 6 row(s)
```

Trino apparently does not interpret column comments.

```
trino> describe delta.default.t1;
 Column |  Type   | Extra | Comment 
--------+---------+-------+---------
 x      | integer |       |         
 y      | varchar |       |         
 z      | varchar |       |         
(3 rows)

Query 20220513_114236_00002_2rhtw, FINISHED, 1 node
Splits: 11 total, 11 done (100.00%)
0.38 [3 rows, 148B] [7 rows/s, 385B/s]
```

On the transaction log file the comment is to be found on the `metadata` field:

```
[root@hadoop-master /]# hdfs dfs -cat /user/hive/warehouse/t1/_delta_log/00000000000000000003.json
{"commitInfo":{"timestamp":1652442096774,"operation":"ADD COLUMNS","operationParameters":{"columns":"[{\"column\":{\"name\":\"z\",\"type\":\"string\",\"nullable\":true,\"metadata\":{\"comment\":\"z comment\"}}}]"},"readVersion":2,"isBlindAppend":true,"operationMetrics":{}}}
{"metaData":{"id":"7644697d-5e28-494a-8b6f-2f48e113eb7c","format":{"provider":"parquet","options":{}},"schemaString":"{\"type\":\"struct\",\"fields\":[{\"name\":\"x\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"y\",\"type\":\"string\",\"nullable\":true,\"metadata\":{}},{\"name\":\"z\",\"type\":\"string\",\"nullable\":true,\"metadata\":{\"comment\":\"z comment\"}}]}","partitionColumns":[],"configuration":{},"createdTime":1652441899688}}
```

Reply:
Thanks for providing the helpful information. Let me leave the comment support to #12424.